pkg/kapp/cmd/app/deploy.go

```diff
@@ -178,6 +178,8 @@ func (o *DeployOptions) Run() error {
 	}
 
 	if o.DiffFlags.Run || hasNoChanges {
+		defer o.writeAppMetadataToFile(app)
```

Comment:
Do we need to defer here?

Looks like we return after this no matter what. No room for erroring.

Reply:
Makes sense, updated.